README.mdx

```diff
@@ -71,6 +71,71 @@ Calimero Network's story is one of relentless pursuit of a more private,
 efficient, and user-centric decentralized web, setting a new standard for what
 decentralized platforms can achieve.
 
+## Architecture
+
+```mermaid
+graph TD
+
+    S2[CLI]
+
+    subgraph Datastore
+        DS1[Context]
+        DS2[Generic]
+        DS3[Network]
+        DS1 --> DS4[Identity]
+        DS1 --> DS5[State]
+        DS1 --> DS6[Transactions]
+        DS2 --> DS7[Scope]
+        DS2 --> DS8[Fragment]
+        DS3 --> DS9[KAD]
+
+        subgraph \n\n
+          DI1[Memory]
+          DI2[S3]
+          DI3[Level]
+          DI4[RocksDB]
+        end
+    end
+
+    subgraph Blobstore
+        BS3[Applications]
+        BS4[Generic Data]
+        subgraph \n
+          BS1[FS]
+          BS2[S3]
+       end
+    end
+    Blobstore --> BS3
+
+    Node --> Network
+
+    subgraph Client
+        C1[C-API]
+    end
+
+    subgraph Server
+        S1[Auth Gated Client]
```

Comment:
this isn't inside server, it's own thing

- `server` depends on `auth gated client`
- `CLI` depends on `server` and `auth gated client`
- `auth gated client` depends on `client`
- `client` depends on `node`
- `scope` depends on `fragment`

Reply:
`C-API` depends on `client`

Reply:
If you can represent this in mermaid:

- `node` depends on `context` and `network` inside `datastore`
- `auth gated client` depends on `generic` inside `datastore`